cli/src/pcluster/templates/cdk_builder.py

```diff
@@ -39,15 +41,19 @@ def build_cluster_template(
 
         LOGGER.info("CDK import completed successfully")
         LOGGER.info("Starting CDK template generation...")
-        with tempfile.TemporaryDirectory() as tempdir:
+        with tempfile.TemporaryDirectory() as cloud_assembly_dir:
             output_file = str(stack_name)
-            app = App(outdir=str(tempdir))
+            app = App(outdir=str(cloud_assembly_dir))
             ClusterCdkStack(app, output_file, stack_name, cluster_config, bucket, log_group_name)
-            app.synth()
-            generated_template = load_yaml_dict(os.path.join(tempdir, f"{output_file}.template.json"))
+            cloud_assembly = app.synth()
+
+            cdk_assets_upload_manager = CDKAssetsManager(CDKV1ClusterCloudAssembly(cloud_assembly))
+            assets_metadata = cdk_assets_upload_manager.upload_assets(bucket=bucket)
```

Comment:
can the CDKAssetsManager be used to return also the "generated_template"? Can we consider the main template as an asset as well?